capa/features/extractors/dnfile/helpers.py

```diff
@@ -190,23 +175,36 @@ def get_dotnet_managed_method_names(pe: dnfile.dnPE) -> Iterator[Tuple[int, str]
             TypeNamespace (index into String heap)
             MethodList (index into MethodDef table; it marks the first of a continguous run of Methods owned by this Type)
     """
-    if not is_dotnet_table_valid(pe, "TypeDef"):
-        return
-
-    for row in pe.net.mdtables.TypeDef:
+    for row in iter_dotnet_table(pe, "TypeDef"):
         for index in row.MethodList:
-            # like File::OpenRead
-            name = f"{row.TypeName}::{index.row.Name}"
-
-            # ECMA II.22.37: TypeNamespace can be null or non-null
-            if row.TypeNamespace:
-                # like System.IO.File::OpenRead
-                name = f"{row.TypeNamespace}.{name}"
-
             token = calculate_dotnet_token_value(index.table.number, index.row_index)
 
-            yield token, name
+            yield token, row.TypeNamespace, row.TypeName, index.row.Name
 
 
 def is_dotnet_mixed_mode(pe: dnfile.dnPE) -> bool:
     return not bool(pe.net.Flags.CLR_ILONLY)
+
+
+def format_dotnet_methodname(namespace: str, class_: str, method: str) -> str:
```

Comment:
if we add a class this could be done via `__str__` or a custom format function

Reply:
see above.